File: PathoFroh/src/main/java/com/patho/main/model/dto/json/JSONPatientMapper.java
package com.patho.main.model.dto.json;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;
import java.util.Date;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.patho.main.model.person.Contact;
import com.patho.main.model.person.Person;
import com.patho.main.model.patient.Patient;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@JsonIgnoreProperties(ignoreUnknown = true)
public class JSONPatientMapper {

	private String titel;
	private String name;
	private String vorname;
	private String geburtsdatum;
	private String ort;
	private String land;
	private String plz;
	private String anschrift;
	private String tel;
	private String weiblich;
	private String piz;
	private String krankenkasse;

	private String error;

	public Patient getPatient() {
		return getPatient(new Patient(new Person(new Contact())));
	}

	public Patient getPatient(Patient patient) {

		patient.getPerson().setTitle(getTitel());
		patient.getPerson().setLastName(getName());
		patient.getPerson().setFirstName(getVorname());

		if (getGeburtsdatum() != null) {
			DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd");
			try {
				LocalDate date = LocalDate.parse(getGeburtsdatum(),format);
				patient.getPerson().setBirthday(date);
			} catch (DateTimeParseException e) {
				patient.getPerson().setBirthday(LocalDate.now());
			}
		} else {
			patient.getPerson().setBirthday(LocalDate.now());
		}

		patient.setInsurance(getKrankenkasse());
		patient.setPiz(getPiz());
		patient.getPerson().getContact().setTown(getOrt());
		patient.getPerson().getContact().setPostcode(getPlz());
		patient.getPerson().getContact().setStreet(getAnschrift());

		patient.getPerson().getContact().setEmail("");
		patient.getPerson().getContact().setPhone(getTel());

		// patient.getPerson().setGender(getWeiblich().equals("1") ?
		// Person.Gender.FEMALE : Person.Gender.MALE);

		return patient;
	}
}

File: src/main/kotlin/com/patho/main/model/json/JSONPatientMapper.kt
package com.patho.main.model.json

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.patho.main.model.patient.Patient
import com.patho.main.model.person.Contact
import com.patho.main.model.person.Person
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeParseException

@JsonIgnoreProperties(ignoreUnknown = true)
class JSONPatientMapper {
    var titel: String? = null
    var name: String? = null
    var vorname: String? = null
    var geburtsdatum: String? = null
    var ort: String? = null
    var land: String? = null
    var plz: String? = null
    var anschrift: String? = null
    var tel: String? = null
    var weiblich: String? = null
    var piz: String? = null
    var krankenkasse: String? = null

    var error: String? = null

    fun getPatient(): Patient {
        return getPatient(Patient(Person(Contact())))
    }

    fun getPatient(patient: Patient): Patient {

        patient.person.title = titel ?: ""
        patient.person.lastName = name ?: ""
        patient.person.firstName = vorname ?: ""

        if (geburtsdatum != null) {
            val format = DateTimeFormatter.ofPattern("yyyy-MM-dd")
            try {
                val date = LocalDate.parse(geburtsdatum, format)
                patient.person.birthday = date
            } catch (e: DateTimeParseException) {
                patient.person.birthday = LocalDate.now()
            }

        } else {
            patient.person.birthday = LocalDate.now()
        }

        patient.insurance = krankenkasse ?: ""
        patient.piz = piz ?: ""
        patient.person.contact.town = ort ?: ""
        patient.person.contact.postcode = plz ?: ""
        patient.person.contact.street = anschrift ?: ""

        patient.person.contact.email = ""
        patient.person.contact.phone = tel ?: ""

        // patient.getPerson().setGender(getWeiblich().equals("1") ?
        // Person.Gender.FEMALE : Person.Gender.MALE);

        return patient
    }
}
